exam (#36)
romankuchin/ica0002-2021: exam/README.md

Each application stack machine should have these services set up and running:
 - HAProxy
 - Keepalived
 - Agama in the Docker container
 - MySQL (master on one machine, replica on the another)
 - Bind slave
 - Prometheus exporters for HAProxy, Keepalived, MySQL and Bind slave

Remaining machine should have these services set up and running:
 - Bind master
 - InfluxDB
 - Telegraf
 - Prometheus
 - Grafana
 - Nginx as frontend for Grafana and Prometheus

Differences with previous tasks
-------------------------------

We tried to make this infrastructure as similar as possible to what you did
during the course, however there are a few differences worth pointing out.

There is one Bind master and _two_ Bind slaves now; you don't need to add master
address to `/etc/resolv.conf` -- only both slave addresses. For example, in this
setup:

	192.168.42.35   # Bind slave
	192.168.42.87   # Bind slave
	192.168.42.124  # Bind master

`/etc/resolv.conf` could look like this:

	nameserver 192.168.42.35
	nameserver 192.168.42.87
	search mydomain.tld

Also, as there are three machines now, so your Grafana dashboards, backup
scripts and documentation should reflect and support it.

Task
----

> Write the Ansible playbook named `infra.yaml` to deploy the infrastrusture

Reviewer: ```suggestion
Update your Ansible playbook named `infra.yaml` to deploy the infrastructure
```

described above.

Infrastructure should be fully set up and online by running exactly this
command:

	ansible-playbook infra.yaml

> Note: it is still allowed to set up MySQL replication manually as we didn't show
> how to set it up with Ansible.
> 

Reviewer: ```suggestion
```

> Provide the backup SLA document for the entire infrastructure (`backup_sla.md`)

Reviewer: ```suggestion
Provide the backup SLA document (`backup_sla.md`)
```

and backup restore instructions for every service that is being backed up
(`backup_restore.md`).

> **Be ready to rebuild your infrastructure from scratch on empty instances!**
> 
> 

Reviewer: ```suggestion

```
